### ISSUE TITLE
Setting glyph color to None results in an array of NaN's

### ISSUE THREAD

**Issue by almarklein** [bokeh/bokeh#2525]
In the example `file/color_scatter.py` we do:
```python
p.scatter(x,y, radius=radii, fill_color=colors, fill_alpha=0.6, line_color=None)
```
Inspecting in JS:
```
> g.visuals.line.color.fixed_value
null
> g.visuals.line.cache.color_array
Array [ NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN, 3990 more… ]
```

Which can be understood from the source in properties.coffee:
```coffeescript
warm_cache: (source, attrs) ->
    for attr in attrs
      prop = @[attr]
      if prop.fixed_value?
        @cache[attr] = prop.fixed_value
      else
        @cache[attr+"_array"] = prop.array(source)
```

What would be the correct way to denote "no coloring"? Can we reserve `null` for that and change `properties.coffee` to: `if prop.fixed_value == undefined`?

**Comment by bryevdv** [comment]
Oh, yikes, I thought this would get interpreted as a `fixed_value`. This should be improved so that happens when a color is set to `None` (that is the canonical way to indicate "no coloring"

**Comment by bryevdv** [comment]
Also, your proposal seems good at a glance, but this is fairly low-level protocol machinery, we will just have to be very careful and test etxtensively (as well as add tests to cover this case)

**Comment by birdsarah** [comment]
Is this also related to #632 (which I closed because at least there's now a warning which was what was asked for) - in that case - see http://nbviewer.ipython.org/github/birdsarah/bokeh-miscellany/blob/master/tests/632_wierd_behavior_when_less_colors_than_points.ipynb - the final color is just filled in for the rest of the array that was missing data.

**Comment by almarklein** [comment]
@birdsarah I think this is different. In this case the color value of `null` gets interpreted as "he, there is no fixed value, so there must be an array of values". Upon getting the array data, you get a lot of `NaN`'s, because there is no such arrays.

**Comment by almarklein** [comment]
@bryevdv can you please help put together a test for this use-case? I've been trying to add something to `test_properties.coffee`, but I fail to get a `ContextProperties` instance to work with.

**Comment by bryevdv** [comment]
sure, probably this weekend some time

**Comment by canavandl** [comment]
This was resolved by PR #3846 

now ``line_color=None`` will serialize as ``'line_color': {'value': None}}``

I'm going to mark as resolved. Please reopen if you feel the issue isn't resolved.
